tests/test_url.py

```diff
@@ -767,6 +767,101 @@ def test_div_with_dots():
     assert url.raw_path == "/path/to"
 
 
+# joinpath
+
+
+def test_joinpath_root():
```

Comment:
Most of the tests added here can be collapsed with the use of `@pytest.mark.parametrize`. Could you implement this? For the output readability, it's best to declare human-readable test IDs.